src/WebUI/Components/ApiKeyDialog.razor

```diff
@@ -24,6 +25,10 @@
                 <MudText Style="font-style: italic">Slow and smart</MudText>
             }
             <MudTextField Disabled="!UsingByoApiKey" InputType="@PasswordInput" Adornment="Adornment.End" AdornmentIcon="@PasswordInputIcon" OnAdornmentClick="ToggleShowPassword" @bind-Value="ApiKeyString" OnKeyDown="ResetValidationErrors" Label="Enter your key" Variant="Variant.Text"></MudTextField>
+            @if (UsingByoApiKey)
+            {
+                <MudCheckBox @bind-Checked="@saveAPIKeyToLocalStorage"  Label="Save API key to LocalStorage"></MudCheckBox>
+            }
```

Comment:
This should be visible, even if the user has not chosen their own API key. This is because, as by my testing - their OpenAI key stays in the local storage after you switch to using SSW's key:
![image](https://github.com/SSWConsulting/SSW.Rules.GPT/assets/20507092/be27b508-6d20-4133-8ea2-aa645fdb35cc)